src/node.cc

```diff
@@ -2192,7 +2192,7 @@ static void WaitForInspectorDisconnect(Environment* env) {
   if (env->inspector_agent()->IsConnected()) {
     // Restore signal dispositions, the app is done and is no longer
     // capable of handling signals.
-#ifdef __POSIX__
+#if defined(__POSIX__) && !defined(NODE_SHARED_MODE)
```

Comment:
Can you provide a bit more detail on why its needed here in addition to the other location.  The section that avoids restoring them because the parent may have changed them makes sense to me. But this part seems more related to the inspector and may still apply in the shared library case.